File: search-service/app/domain/search_request.py
"""
PVK Cinemas Search Service — Domain Models
Authoritative source: SPEC-BASELINE §I, SRS §6.10, SRS §12, ARCH §11.
"""

from typing import Optional
from pydantic import BaseModel, Field, field_validator


class SearchRequest(BaseModel):
    """
    Inbound search query from Spring Boot backend.
    Authority: SRS FR-090, FR-093, FR-094, API-SPEC §23.
    """
    query: str = Field(..., description="Raw user query text")
    city_id: Optional[int] = Field(None, description="Optional city filter")
    max_results: int = Field(20, ge=1, le=50, description="Maximum results to return")

    @field_validator("query")
    @classmethod
    def query_must_not_be_empty(cls, v: str) -> str:
        stripped = v.strip()
        if not stripped:
            raise ValueError("query must not be blank")
        if len(stripped) > 500:
            raise ValueError("query must not exceed 500 characters")
        return stripped

File: backend/src/main/java/com/pvk/cinemas/search/dto/SearchRequest.java
package com.pvk.cinemas.search.dto;

public class SearchRequest {
    private String query;
    private Integer cityId;
    private Integer limit = 10;

    public SearchRequest() {}

    public SearchRequest(String query, Integer cityId, Integer limit) {
        this.query = query;
        this.cityId = cityId;
        this.limit = limit != null ? limit : 10;
    }

    public String getQuery() { return query; }
    public void setQuery(String query) { this.query = query; }

    public Integer getCityId() { return cityId; }
    public void setCityId(Integer cityId) { this.cityId = cityId; }

    public Integer getLimit() { return limit; }
    public void setLimit(Integer limit) { this.limit = limit; }
}
